### PR TITLE
Development and production mode for Artik devices

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Added functionality for switching Artik profiles.
Improved resource monitoring system.

### CODE REVIEW

**Comment by vitaliy-guliy** [comment]
screenshot from 2016-08-12 16 30 16

File: plugins/plugin-artik-ide/src/main/java/org/eclipse/che/plugin/artik/ide/profile/ArtikModeActionFactory.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+/**
+ * Factory for creating actions to switch Artik profile.
+ *
+ * @author Vitaliy Guliy
+ */
+public interface ArtikModeActionFactory {
+
+    /**
+     * Craetes TurnDevelopmentModeAction instance with a given Artik machine name.
```
Comment (ashumilova) [inline plugins/plugin-artik-ide/src/main/java/org/eclipse/che/plugin/artik/ide/profile/ArtikModeActionFactory.java]: Craetes - Creates

**Comment by ashumilova** [comment]
+1

File: plugins/plugin-artik-ide/src/main/java/org/eclipse/che/plugin/artik/ide/profile/ArtikModeActionFactory.java (hunk 1 above)
Comment (vitaliy-guliy) [inline plugins/plugin-artik-ide/src/main/java/org/eclipse/che/plugin/artik/ide/profile/ArtikModeActionFactory.java]: Fixed. Thanks

**Pull request merged**